README.md

```diff
@@ -0,0 +1,38 @@
+# deno_postgres_interop
+[![style: solid](https://img.shields.io/badge/style-solid-orange)](https://pub.dev/packages/solid_lints)
+[![$solid_lints](https://nokycucwgzweensacwfy.supabase.co/functions/v1/get_project_badge?projectName=deno_postgres_interop)](https://www.worklog.ai)
+
+This package was created to be used in supabase edge functions
+because existing db-driver/orm packages depend on dart:io.
+It should also work for other cases with dart2js.
+
+## Usage
+1. Compile the code that uses this package with:
+   ```bash
+   dart run edge build supabase_functions
+   ```
+
+2. Add imports from `postgres v0.17.0` to compiled `main.dart.js`, e.g.:
+   ```js
+   import { Client } from "https://deno.land/x/postgres@v0.17.0/mod.ts";
+   ```
+
+3. Replace all constructors to not use `self.`, e.g.:
+
+   `new self.Pool(...)` => `new Pool(...)`
```

Comment:
Let's see if we can change how gen works to avoid this step. Or maybe some other wrapper class that will fix it.

Reply:
So far it seems that build_runner is necessary
IDK if we want to do that yet?

Reply:
[related](https://github.com/solid-software/deno_postgres_interop/pull/1#discussion_r1319096246)

Reply:
Okay, I found a bit better solution, it still requires manual diff, but it is now of fixed size

we will keep `self.X()` constructor calls, but we will also initialize `self.X` with `X`

so instead of just import, it becomes:
```js
import { Client } from "https://deno.land/x/postgres@v0.17.0/mod.ts";
self.Client = Client;
```
the same will be applied for other classes:
```js
import { Client } from "https://deno.land/x/postgres@v0.17.0/mod.ts";
import { Pool } from "https://deno.land/x/postgres@v0.17.0/mod.ts";
self.Client = Client;
self.Client = Pool;
```

Reply:
I'm not sure that will work either - annotations expand into dart code, so this won't make much sense,
unless we keep annotations there and then we check what classes are used in code - take data from their annotations and insert into js code

it won't run via build_runner tho (if I'm not mistaken)

Reply:
There's a more shallow solution - select all substrings `new self.X(`
and see if the X is from deno_postgres - if yes, add the lines to `main.dart.js`

I think that should be sufficient for now, I'll implement that

Reply:
currently implemented in python, will port to dart